### PR TITLE
R4R: guardian refactor

### PR DESCRIPTION
**Pull request by chengwenxi**
1. add delete handler
2. add profiler/trustee type, `genesis` type account can add/delete `ordinary` account

### CODE REVIEW

File: modules/guardian/genesis.go [hunk 1]
```diff
@@ -6,11 +6,11 @@ import (
...
 // GenesisState - all guardian state that must be provided at genesis
 type GenesisState struct {
-	Profilers []Profiler `json:"profilers"`
-	Trustees  []Trustee  `json:"trustees"`
+	Profilers []Guardian `json:"profilers"`
+	Trustees  []Guardian `json:"trustees"`
 }
 
-func NewGenesisState(profilers []Profiler, trustees []Trustee) GenesisState {
+func NewGenesisState(profilers, trustees []Guardian) GenesisState {
```
Comment (haifengxi) [inline modules/guardian/genesis.go]: At genesis, we should ensure Guardian.AddressType == genesis

File: modules/guardian/msgs.go [hunk 2]
```diff
@@ -33,30 +32,142 @@ func (msg MsgAddProfiler) GetSignBytes() []byte {
...
-	// Must contain alphanumeric characters, _ and - only
-	reg := regexp.MustCompile(`[^a-zA-Z0-9_-]`)
-	return !reg.Match([]byte(name))
+//______________________________________________________________________
+
+type AddGuardian struct {
+	Description  string         `json:"description"`
+	Address      sdk.AccAddress `json:"address"`
+	AddedAddress sdk.AccAddress `json:"added_address"`
+}
```
Comment (haifengxi) [inline modules/guardian/msgs.go]: Description  string
Address       sdk.AccAddress     // address added
AddedBy     sdk.AccAddress     // address that initiated the tx

File: modules/guardian/msgs.go [hunk 3]
```diff
@@ -33,30 +32,142 @@ func (msg MsgAddProfiler) GetSignBytes() []byte {
...
+
+type DeleteGuardian struct {
+	Address        sdk.AccAddress `json:"address"`
+	DeletedAddress sdk.AccAddress `json:"deleted_address"`
+}
```
Comment (haifengxi) [inline modules/guardian/msgs.go]: Address sdk.AccAddress // address deleted
DeletedBy sdk.AccAddress // address that initiated the tx

File: modules/guardian/types.go [hunk 4]
```diff
@@ -2,37 +2,101 @@ package guardian
...
-type Profiler struct {
-	Name      string         `json:"name"`
-	Addr      sdk.AccAddress `json:"addr"`
-	AddedAddr sdk.AccAddress `json:"added_addr"`
+type Guardian struct {
+	Description  string         `json:"description"`
+	AccountType  AccountType    `json:"type"`
+	Address      sdk.AccAddress `json:"address"`
+	AddedAddress sdk.AccAddress `json:"added_address"`
 }
```
Comment (haifengxi) [inline modules/guardian/types.go]: Description string
AccountType AccountType
Address sdk.AccAddress     // this guardian's address
AddedBy sdk.AccAddress     // address that initiated the AddGuardian tx

File: modules/guardian/genesis.go (hunk 1 above)
Comment (chengwenxi) [reply modules/guardian/genesis.go]: Yes, I will add AddressType check.

Comment (chengwenxi) [reply modules/guardian/genesis.go]: We should add a new function to check GenesisState.

File: client/guardian/cli/sendtx.go [hunk 5]
```diff
@@ -28,25 +28,126 @@ func GetCmdCreateProfiler(cdc *codec.Codec) *cobra.Command {
...
 			if len(paStr) == 0 {
-				return fmt.Errorf("must use --profiler-name flag")
+				return fmt.Errorf("must use --address flag")
+			}
+			pAddr, err := sdk.AccAddressFromBech32(paStr)
+			if err != nil {
+				return err
+			}
+			msg := guardian.NewMsgDeleteProfiler(pAddr, fromAddr)
+			cliCtx.PrintResponse = true
```
Comment (HaoyangLiu) [inline client/guardian/cli/sendtx.go]: Remove `cliCtx.PrintResponse = true`

**Pull request merged**